presto-orc/src/main/java/com/facebook/presto/orc/reader/FloatStreamReader.java

```diff
@@ -174,12 +175,180 @@ public String toString()
     public void close()
     {
         systemMemoryContext.close();
-        nullVector = null;
     }
 
     @Override
     public long getRetainedSizeInBytes()
     {
-        return INSTANCE_SIZE + sizeOf(nullVector);
+        return INSTANCE_SIZE;
+    }
+
+    @Override
+    public void setFilterAndChannel(Filter filter, int channel, int columnIndex, Type type)
+    {
+        checkArgument(type == REAL, "Unsupported type: " + type.getDisplayName());
+        super.setFilterAndChannel(filter, channel, columnIndex, type);
```

Comment:
I don't understand why we don't put these `checkArgument` in the super type and not override this.

Reply:
@tdcmeehan Super type doesn't know which types we support. Are you suggesting to pass this on to constructor?

Reply:
Yes

Reply:
@tdcmeehan Actually, this is a bit more complicated. For Decimal reader, the check is `type instanceof DecimalType`. I see two options: (1) leave as is; (2) define abstract `void checkType(type)` method in the base class and call it from `setFilterAndChannel` implementation in the base class. What do you think?